examples/cli-app/src/main.rs

```diff
@@ -0,0 +1,169 @@
+use futures::executor::block_on;
+use lazy_static::lazy_static;
+use meilisearch_sdk::{client::*,settings::Settings};
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use std::fs::File;
+use std::io::{stdin, Read};
+
+// instantiate the client. load it once
+lazy_static! {
+    static ref CLIENT: Client = Client::new("http://localhost:7700", "masterKey");
+}
+
+fn main() {
+    block_on(async move {
+        // build the index
+        build_index().await;
+
+        // enter in search queries or quit
+        loop {
+            println!("Enter a search query or type \"q\" or \"quit\" to quit:");
+            let mut input_string = String::new();
+            stdin()
+                .read_line(&mut input_string)
+                .expect("Failed to read line");
+            match input_string.trim() {
+                "quit" | "q" | "" => {
+                    println!("exiting...");
+                    break;
+                }
+                _ => {
+                    search(input_string.trim()).await;
+                }
+            }
+        }
+        // get rid of the index at the end, doing this only so users don't have the index without knowing
+        let _ = CLIENT.delete_index("clothes").await.unwrap();
+    })
+}
+
+async fn search(query: &str) {
+    // make the search query, which excutes and serializes hits into the
+    // ClothesDisplay struct
+    let query_results = CLIENT
+        .index("clothes")
+        .search()
+        .with_query(query)
+        .execute::<ClothesDisplay>()
+        .await
+        .unwrap()
+        .hits;
+
+    // display the query results
+    if query_results.is_empty() {
+        println!("no results...");
+    } else {
+        for clothes in query_results {
+            let display = clothes.result;
+            println!("{}", format_args!("{}", display));
+        }
+    }
+}
+
+async fn build_index() {
+    // reading and parsing the file
+    let mut file = File::open("./assets/clothes.json").unwrap();
+    let mut content = String::new();
+    file.read_to_string(&mut content).unwrap();
```

Comment:
```suggestion
    let content = include_str!("./assets/clothes.json");
```

Hey, yesterday I was thinking it could be cool to use this macro instead of loading the file ourselves. 
This ensures **at compile-time** that the dataset is available thus the example should never break 👍

Reply:
Thats neat, I didn't know that macro existed. I made the macro change and did some clippy cleanup